app/src/main/java/nu/yona/app/api/manager/impl/ActivityCategoryManagerImpl.java

```diff
@@ -52,33 +53,13 @@ public ActivityCategoryManagerImpl(Context context) {
      * @param listener
      */
     @Override
-    public void getActivityCategoriesById(final DataLoadListener listener) {
-        try {
-            User user = YonaApplication.getEventChangeManager().getDataState().getUser();
-            if (user != null && user.getLinks() != null && user.getLinks().getYonaAppActivity() != null && !TextUtils.isEmpty(user.getLinks().getYonaAppActivity().getHref())) {
-                activityCategoriesNetwork.getActivityCategories(authenticateDao.getUser().getLinks().getYonaAppActivity().getHref(), new DataLoadListener() {
-                    @Override
-                    public void onDataLoad(Object result) {
-                        updateActivityCategories(result, null);
-                        if (listener != null) {
-                            listener.onDataLoad(result);
-                        }
-                    }
-
-                    @Override
-                    public void onError(Object errorMessage) {
-                        if (listener != null) {
-                            listener.onError(errorMessage);
-                        }
-                    }
-                });
-            } else {
-                if (listener != null) {
-                    listener.onError(new ErrorMessage(mContext.getString(R.string.urlnotfound)));
-                }
-            }
-        } catch (Exception e) {
-            AppUtils.throwException(ActivityCategoryManagerImpl.class.getSimpleName(), e, Thread.currentThread(), listener);
+    public void getActivityCategoriesById(DataLoadListener listener) {
+        User user = YonaApplication.getEventChangeManager().getDataState().getUser();
+        if (user != null && user.getLinks() != null && user.getLinks().getYonaAppActivity() != null && !TextUtils.isEmpty(user.getLinks().getYonaAppActivity().getHref())) {
+            DataLoadListenerImpl listenerWrapper = new DataLoadListenerImpl((result) -> updateActivityCategories(result, null), listener);
+            activityCategoriesNetwork.getActivityCategories(authenticateDao.getUser().getLinks().getYonaAppActivity().getHref(), listenerWrapper);
+        } else if (listener != null) {
+            listener.onError(new ErrorMessage(mContext.getString(R.string.urlnotfound)));
```

Comment:
For this one, I would make a helper method on ``DataLoadListenerImpl``, so you can change this line and the above one into something like ``DataLoadListenerImpl.reportError(listener, R.string.urlnotfound);``